File: clients/typescript/examples/collections-count-all.ts
import {
  DataAPIClient,
  TooManyDocumentsToCountError,
} from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

(async function () {
  try {
    // Count documents
    const result = await collection.countDocuments({}, 500);

    console.log(result);
  } catch (error) {
    if (error instanceof TooManyDocumentsToCountError) {
      console.log("Number of documents exceeds upper bound or API limit");
    } else {
      throw error;
    }
  }
})();

File: clients/csharp/examples/collections-count-all.cs
using DataStax.AstraDB.DataApi;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Count documents
    var result = await collection.CountDocumentsAsync(null, 500);

    Console.WriteLine(result);
  }
}
